thrift/thrift0.13/src/main/java/com/linecorp/armeria/internal/server/thrift/ThriftDocServicePlugin.java

```diff
@@ -188,18 +188,24 @@ ServiceInfo newServiceInfo(Class<?> serviceClass, Iterable<EndpointInfo> endpoin
     }
 
     @Nullable
-    private MethodInfo newMethodInfo(Method method, Iterable<EndpointInfo> endpoints,
+    private MethodInfo newMethodInfo(Method method, Entry entry,
                                      DocServiceFilter filter) {
-        final String methodName = method.getName();
+        // Get the function name in thrift IDL
+        final String thriftFunctionName;
+        if (entry.thriftServiceEntry != null) {
+            thriftFunctionName = entry.thriftServiceEntry.functionName(method.getName());
+        } else {
+            thriftFunctionName = method.getName();
```

Comment:
Sorry 😅   Can you add a comment that says "this logic exists for tests"?

```suggestion
            // entry.thriftServiceEntry can be null for tests
            thriftFunctionName = method.getName();
```